ndscan/plots/model/online_analysis.py

```diff
@@ -100,25 +104,28 @@ async def _recompute_fit(self):
         y_errs = self._source_data.get("y_err", None)
 
         loop = asyncio.get_event_loop()
-        self._last_fit_params, self._last_fit_errors = await loop.run_in_executor(
-            self._fit_executor, _run_fit, self._fit_type, xs, ys, y_errs,
-            self._constants, self._initial_values)
+
+        self._fit_obj.x = xs
+        self._fit_obj.y = ys
+        self._fit_obj.y_err = y_errs
+
+        fit = await loop.run_in_executor(self._fit_executor, _run_fit, self._fit_obj)
+        self._last_fit_params, self._last_fit_errors = fit
 
         self._recompute_in_progress = False
         self.updated.emit()
 
 
-def _run_fit(fit_type, xs, ys, y_errs, constants, initial_values):
+def _run_fit(fit_obj):
```

Comment:
Does this actually work? I kept the interface painfully scalar since I figured that the arguments would need to be transferred out of process to the executor somehow, but perhaps the fit object is pickled/unpickled just fine?

Reply:
yes! The only thing that doesn't work are the fit result properties that I dynamically create during `__init__` (and, in any case, I'm not sure that those are really good design anyway).

....although I'm still not convinced that this is the right way of handling it. I was thinking of changing this to pass the `FitDescription` into the executor and then have the code here do exactly the same thing as the applet.